host/volume/api/http.go

```diff
@@ -0,0 +1,114 @@
+package volumeapi
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/julienschmidt/httprouter"
+	"github.com/flynn/flynn/host/volume"
+	"github.com/flynn/flynn/host/volume/zfs"
+	"github.com/flynn/flynn/pkg/httphelper"
+	"github.com/flynn/flynn/pkg/random"
+)
+
+type HttpAPI struct {
+	vman *volume.Manager
+}
+
+func NewHttpAPI(vman *volume.Manager) *HttpAPI {
+	return &HttpAPI{vman: vman}
+}
+
+func RegisterRoutes(api *HttpAPI, r *httprouter.Router) {
+	r.POST("/volume/provider", api.CreateProvider)
+	r.POST("/volume/provider/:provider_id/newVolume", api.Create)
+	r.PUT("/volume/instances/:id/snapshot", api.Snapshot)
+	r.GET("/volume/instances/:id/inspect", api.Inspect)
```

Comment:
I agree with having volumes under the top level, however I'd still follow the shape of the rest of the API and make the operations more RESTful:

```
GET /volumes/providers - list providers
POST /volumes/providers - create provider
GET  /volumes/providers/:provider_id/volumes - list volumes for provider
POST /volumes/providers/:provider_id/volumes - create volume
GET  /volumes/:volume_id - get volume details
PUT  /volumes/:volume_id/snapshots - create snapshot
```

Namespacing providers under `/volumes` directly might end up a bit ugly, so we could top-level them and namespace the entire API under `/data` or `/storage` (`/volume/volumes/:id` seems a bit ugly) -- `GET /storage/providers`:

```
GET  /storage/providers - list providers
POST /storage/providers - create provider
GET  /storage/providers/:provider_id/volumes - list volumes for provider
POST /storage/volumes/providers/:provider_id/volumes - create volume
GET  /storage/volumes/:volume_id - get volume details
PUT  /storage/volumes/:volume_id/snapshots - create snapshot
```